### PR TITLE
Adding stress_devnet script that spams transactions to local devnet

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Also as part of this commit updating local typescript files (from one of the latest fiddles):
- decoder.ts
- encoder.ts
- near.ts

### CODE REVIEW

**Comment by nearmax** [comment]
Is it able to crash the DevNet?

**Comment by nearmax** [comment]
I ran it but it returns the following:
https://www.dropbox.com/s/grhns4pt2vyq9oc/Screenshot%202019-02-21%2010.46.45.png?dl=0

**Comment by evgenykuzyakov** [comment]
Seems something is wrong with the keystore for alice key. Maybe need to restart the devnet after you regenerated a key from the new storage.

It doesn't crash the devnet, just puts heavy usage. I think it's blocked by the read RPC to get account nonce which prevents devnet from beign overloaded with new transactions.

**Comment by nearmax** [comment]
> Seems something is wrong with the keystore for alice key. Maybe need to restart the devnet after you regenerated a key from the new storage.
Still does not work, see: https://www.dropbox.com/s/pseh7ucdxqgob8c/stresstest.mov?dl=0

> It doesn't crash the devnet, just puts heavy usage. I think it's blocked by the read RPC to get account nonce which prevents devnet from beign overloaded with new transactions.
In general, stress test is a great idea, because then we can:
1) Fix our DevNet to prevent crashing under load;
2) Test its throughput, e.g. a) at what rate of RPC per second it starts losing transactions and what percentage. Also test the transaction throughput delay, e.g. b) after how many seconds transaction makes it into the block, c) how many it takes for complex contract with receipts to get fully complete. Ideally we would want to even plot (a),(b), and (c) with x-axis being RPC per seconds;

Stresstest that spams DevNet with RPCs but does not achieve the crash or measures the throughput is hard to act on, because it is always successful (unless someone submits a breaking PR, but then integration tests should catch it).

**Comment by evgenykuzyakov** [comment]
I agree. I just don't want to lose the work I did already. It's not integrated with any automatic tests yet, more like a script that you can run.

Removed auto-closing of the issue. My goal for this script was to crash the current devnet. Probably need to update the script to actually achieve it.

**Comment by nearmax** [comment]
Also, yesterday we talked with @ilblackdragon and he had a great idea. Several parts of our infra are required to be tested for similar scenarios, e.g. a scenario could be: create two accounts, deploy a contract, send transactions to the contract from both accounts, verify execution result through view calls. These scenarios are already tested in several places:
* Runtime;
* Benchmarks for Client (I will make integration tests our of them soon);
* Integration tests for DevNet in `test_rps_cli.rs`;
* Integration test for TestNet in `test_testnet.rs`;

We would also like to test ShardChain for such scenarios. @ilblackdragon had an idea that we need to take [an existing tool, `User`](https://github.com/nearprotocol/nearcore/blob/87a750d2c731df323c5be942642cb051140a0309/node/runtime/src/test_utils.rs#L165) and make it compatible with all these scenarios. `User` would have the API that would allow one to create+deploy contract, call it multiple times, verify results through view calls. We can then abstract out its dependence so it depends on some a trait instead of a Runtime. Then we would implement this trait for Runtime, ShardChain, Client, DevNet, TestNet, etc.

This stress test seem to fall in the same category, and would benefit from using User. So it would make sense to have it written in Rust. Do you think it would make sense to convert it to Rust in this or next PR?

**Comment by evgenykuzyakov** [comment]
The idea for this stress test is to test release version of devnet. It would make sense to use nearlib instead of relying on rpc.py since it hold the account and keys in memory. I should probably rewrite it that way.
Simplifying it using traits is not going to cover some other issues, e.g. working with real storage or HTTP interface or some other things.

File: scripts/stress_devnet.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import json
+from time import sleep
+from subprocess import call
+
+def setup(account_id):
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    os.chdir(os.path.join(dir_path, '..'))
+    # Deploying a contract
+    call(("./scripts/rpc.py", "create_account", "test_contract", "0"))
+    sleep(3.1)
```
Comment (nearmax) [inline scripts/stress_devnet.py]: We need to avoid timers

**Comment by nearmax** [comment]
> It would make sense to use nearlib instead of relying on rpc.py since it hold the account and keys in memory. I should probably rewrite it that way.
It is a good idea, I think we should be converging on one language when it comes to interacting with RPC whether it is tests or nearlib, for maintenance reasons. JS is an obvious choice over Python, because nearlib is in JS.

> Simplifying it using traits is not going to cover some other issues, e.g. working with real storage or HTTP interface or some other things.
We might be thinking about different approaches We can discuss it in detail in person some time later.

**Comment by ilblackdragon** [comment]
@evgenykuzyakov what's happening here? Should we submit it at the end or we have a better plan for stress testing?

**Comment by evgenykuzyakov** [comment]
It doesn't work anymore. Needs to update to pynear.

**Pull request closed without merging**